fortio_main.go

```diff
@@ -41,11 +41,17 @@ var httpOpts fhttp.HTTPOptions
 type flagList struct {
 }
 
-// Unclear when/why this is called and necessary
+// String is the method to format the flag's value, part of the flag.Value interface.
+// The String method's output will be used in diagnostics.
 func (f *flagList) String() string {
 	return ""
 }
+
+// Set is the method to set the flag value, part of the flag.Value interface.
+// Set's argument is a string to be parsed to set the flag.
+// It's a comma-separated list, so we split it.
 func (f *flagList) Set(value string) error {
+	httpOpts.Init(value)
```

Comment:
this is wrong location for the fix

please put init in http.go and a test in http_test.go
```
func TestGetHeaders(t *testing.T) {
	o := NewHTTPOptions("")
	o.AddAndValidateExtraHeader("FOo:baR")
```
replace first line by
```
        o := HTTPOptions{}
```